internal_ws/ykbh/src/lib.rs

```diff
@@ -194,6 +203,8 @@ impl SIRInterpreter {
     /// Run the SIR interpreter after it has been initialised by a guard failure. Since we start in
     /// the block where the guard failed, we immediately skip to the terminator and interpret it to
     /// see which block we need to start interpretation in.
+    /// XXX we have one function called "interpret" and another "_interpret" -- both public! At
+    /// least one of them needs a better name.
```

Comment:
Does `_interpret` need to be public? Try removing `pub`.

Reply:
Note: we should never have functions beginning with `_`!

Reply:
`_interpret` is used by the test suite so it has to be public. But we can certainly do something about the name.

Reply:
How about we name rename this to `interpret_with_ctx` and then `_interpret` becomes `interpret`.

Reply:
That's better but I'm not sure what it means to "interpret without context", so that's still maybe not perfect. Perhaps better doc strings can clear this up, however.

Reply:
@ptersilie  how about `inner_interpret` or `interpret_impl`?

Reply:
There is no "interpret without context" as `interpret_with_ctx` calls `interpret`. But I agree, this is a bit confusing. `_interpret` only ever used for the stand-alone `SIRInterpreter` tests. Maybe we can update the tests to use the same function. The main difference is the `self.terminator()` line, which isn't needed when testing the Sir interpreter without blackholing.

Reply:
There's a temporary solution in df14c2a, renaming them to `sg_interpret` (stopgap_interpret) and `interpret`. Though I have an idea how to get rid of one of the functions entirely, though its a little more involved and thus for a later PR.

Reply:
Refactored in #266.